zml/aio/torch.zig

```diff
@@ -312,102 +312,97 @@ pub const PickleData = struct {
                 try self.parseValue(allocator, store, prefix, build.args);
             },
             .pers_id => |pers_id| try self.parseValue(allocator, store, prefix, pers_id.ref),
-            .seq => |*seq| switch (seq[0]) {
-                .list, .tuple, .set, .frozen_set => {
-                    const elemCheck = struct {
-                        fn call(comptime T: ValueType) fn (v: Value) bool {
-                            return struct {
-                                fn call(val: Value) bool {
-                                    return val == T;
+            .seq => |seq| {
+                const seq_type, const seq_values = seq;
```

Comment:
actually why is seq a tuple and not a struct in the first place ? that would make this code easier to read.

Reply:
Yea, there's no reason this can't be a struct with explicit fields. Will be fixed in the next commit.